react/jest.config.cjs

```diff
@@ -179,7 +177,7 @@ module.exports = {
   // timers: "real",
 
   // A map from regular expressions to paths to transformers
-  transform: {'^.+\\.(js|jsx)?$': 'babel-jest'},
```

Comment:
is this change needed?

Reply:
The unit test for MainMenu was failing with a TypeError: Cannot read properties of undefined (reading 'createContext') error before I added that transformation for the ts and tsx file type. React was already being imported in MainMenu too, but not being read properly before apparently

Reply:
Perhaps that bug occurred because of the (beneficial, I think) change of —

```diff
-        "test": "jest --testPathIgnorePatterns='<rootDir>/src/core-components/'",
+        "test": "jest",
```

— in `react/package.json`. Instead of ignoring, they must… now be transformed?

Reply:
Oh, I maybe morning-muddle-minded. I read the full PR description and more of the diff, now. There is other possibly relevant code.

Reply:
Yeah MainMenu doesn't fall under the core-components umbrella, and I just moved the testPathIgnorePatterns line to the jest.config file instead. I'm not quite sure why ts and tsx file need to be transformed now, but weren't before. But looking at the diff of package-lock.json there's some babel items that were upgraded, so it could be related to that